### PR TITLE
Finished form class refactor. Added validation (built form fields made on request) and a way to get the validated data only.

### PR DESCRIPTION
**Pull request by LCD344**

### CODE REVIEW

File: package/CoopCms/src/app/Forms/FormFields/PasswordField.php [hunk 1]
```diff
@@ -12,4 +12,14 @@
...
 {
     protected $viewName = 'coopCms.formGenerator.fields.passwordField';
+	
+	public function validationRules() {
+		$rules = parent::validationRules();
+		
+		$rules[] = 'string';
+		$rules[] = 'nullable';
+		$rules[] = 'confirmed';
+		return $rules;
```
Comment (Cucurrucho) [inline package/CoopCms/src/app/Forms/FormFields/PasswordField.php]: password shouldn't always confirm

File: package/CoopCms/src/app/Forms/FormFields/TextField.php [hunk 2]
```diff
@@ -10,5 +10,13 @@
...
 
 class TextField extends BaseField {
-    protected $viewName = 'coopCms.formGenerator.fields.textField';
+	protected $viewName = 'coopCms.formGenerator.fields.textField';
+	
+	public function validationRules() {
+		$rules = parent::validationRules();
+		
+		$rules[] = 'string';
+		$rules[] = 'nullable';
```
Comment (Cucurrucho) [inline package/CoopCms/src/app/Forms/FormFields/TextField.php]: Why is it nullable on default?

File: package/CoopCms/src/app/Forms/FormFields/TextareaField.php [hunk 3]
```diff
@@ -11,4 +11,13 @@
 
 class TextareaField extends BaseField {
 	protected $viewName = 'coopCms.formGenerator.fields.textareaField';
+	
+	
+	public function validationRules() {
+		$rules = parent::validationRules();
+		
+		$rules[] = 'string';
+		$rules[] = 'nullable';
```
Comment (Cucurrucho) [inline package/CoopCms/src/app/Forms/FormFields/TextareaField.php]: Why is it nullable on default?

File: package/CoopCms/src/app/Forms/FormFields/TextField.php (hunk 2 above)
Comment (LCD344) [reply package/CoopCms/src/app/Forms/FormFields/TextField.php]: string for strings - nullable to allow it be empty

File: package/CoopCms/src/app/Forms/FormFields/TextareaField.php (hunk 3 above)
Comment (LCD344) [reply package/CoopCms/src/app/Forms/FormFields/TextareaField.php]: string for strings - nullable to allow it be empty

**Pull request merged**